### ISSUE TITLE
🐛 [Bug] Compile error

### ISSUE THREAD

**Issue by Err0r1** [alexkulya/pandaria_5.4.8#57]
## 🐛 Bug report

- No information
![screenshot 3199](https://user-images.githubusercontent.com/74456323/167936077-87f500a9-3d21-40a4-9666-02fab7a5c471.png)

### Expected behavior

- No error..

### Steps to reproduce

1. Install debian 11
2. Try to compile
3. ...
4. ...
5. Profit.

### Version

Client Version: 
- No information

**Comment by pangolp** [comment]
The error is simple, it says that the data type is a uint32, that is, a positive integer (unsigned) and inside the case, they are using a -1. So clearly it is not going to work. I don't currently have a Linux machine, to compile, but it would be fixed by modifying that. The strange thing is that this error has not happened to anyone else. I haven't compiled the emulator yet.

**Comment by Steyfix** [comment]
hello a solution to the problem because I also find it under ubuntu 20.04 thank you ?

**Comment by pangolp** [comment]
> hello a solution to the problem because I also find it under ubuntu 20.04 thank you ?

In this case, the error is easy to correct. I give you a clue to the solution. It turns out that the data type is `uint32`, that is, it is an integer, which cannot have negative numbers. However, in the case, it evaluates the possibility that it is -1. You have 2 alternatives, or you change the data type to `int32` so that it admits positive and negative or analyzing the case, you can delete the option of -1, and replace it by a default. But it all depends on what the switch does, and the type of data you have to return.

**Comment by Steyfix** [comment]
Hello and thank you for your answer with a nice explanation. So I changed the default value and the compilation passed. thanks again

**Comment by davidwow1024** [comment]
Could you upload the commit so that everyone who uses Linux can compile and thanks in advance

**Comment by Steyfix** [comment]
yes I'm going to place the commit once I'm sure that with a make install the server launches well ^^

**Comment by davidwow1024** [comment]
Thanks @Steyfix and @pangolp

**Comment by Steyfix** [comment]
hello someone would have a relase of the bdd world because the one provided in the sources has a table generation error on the locales_quest thank you?

**Comment by Steyfix** [comment]
Fix value to change so that the compilation passes under ubuntu

![image](https://user-images.githubusercontent.com/12504219/173901112-179c5b57-bcf0-46b4-b7a4-a7a92acac56a.png)

**Comment by Mistixx** [comment]
[Steyfix](https://github.com/Steyfix)  why don't you answer my messages to your discord unblock me
